tests/fpga/hbm_validation.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+
+from dace.sdfg.validation import InvalidSDFGEdgeError, InvalidSDFGError, InvalidSDFGNodeError, validate
+from dace import subsets as sbs, dtypes, memlet as mem
+import dace
+import numpy as np
+from dace import subsets
+from dace.sdfg import nodes as nd
+
+#A test to check the changes to the validation required for the support for HBM
+
+
+def checkInvalid(sdfg, exceptiontype):
+    ok = False
+    try:
+        sdfg.validate()
+    except exceptiontype as msg:
+        ok = True
+    assert ok
+
+
+def deepscopeTest():
+    @dace.program
+    def deepscope(input: dace.int32[12, 10], output: dace.int32[12, 10]):
+        for k in dace.map[0:10]:
+            for j in dace.map[0:2]:
+                for z in dace.map[0:2]:
+                    with dace.tasklet:
+                        _read << input[k + j + z, 0]
+                        _write >> output[k + j * z, 0]
+                        _write = _read + 1
+
+    sdfg = deepscope.to_sdfg()
+    for state in sdfg.nodes():
+        for node in state.nodes():
+            if isinstance(node, nd.MapEntry):
+                node.map.schedule = dtypes.ScheduleType.Unrolled
+    sdfg.arrays["input"].location["hbmbank"] = subsets.Range.from_string("0:12")
+    sdfg.arrays["output"].location["hbmbank"] = subsets.Range.from_string(
+        "12:24")
+    sdfg.apply_fpga_transformations(validate=False)
+    sdfg.validate()
+
+
+def multitaskletTest():
```

Comment:
```suggestion
def multi_tasklet():
```

We can't name stuff with "test" at the moment because pytest will pick.

Actually -- maybe you _could_ use PyTest for this? Does this even need the FPGA tools to run?

Reply:
No not really, just put it there because it logically belongs to FPGA. Can I just add it to the top level folder and name all functions I want to be tested with "test" at the end?

Reply:
No not really, just put it there because it logically belongs to FPGA. Can I just add it to the top level folder and name all functions I want to be tested with "test" at the end?